File: ios/Tokopedia/Classses/UIRefactor/Search/SearchResult/ProductLabel.m
//
//  ProductLabel.m
//  Tokopedia
//
//  Created by Tonito Acen on 10/24/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

#import "ProductLabel.h"

@implementation ProductLabel
+(RKObjectMapping *)mapping{
    RKObjectMapping *labelMapping = [RKObjectMapping mappingForClass:[ProductLabel class]];
    [labelMapping addAttributeMappingsFromArray:@[@"title", @"color"]];
    return labelMapping;
}

- (NSString*)title {
    return _title ?:@"";
}

@end

File: ios/Tokopedia/Features/UIRefactor/Search/SearchResult/ProductLabel.swift
//
//  ProductLabel.swift
//  Tokopedia
//
//  Created by Ferico Samuel on 5/15/17.
//  Copyright © 2017 TOKOPEDIA. All rights reserved.
//

import Foundation
import Unbox
import RestKit

@objc(ProductLabel)
final class ProductLabel:NSObject, Unboxable {
    var title:String = ""
    var color:String = ""
    
    static func mapping() -> RKObjectMapping {
        let mapping : RKObjectMapping = RKObjectMapping(for: self)!
        mapping.addAttributeMappings(from: ["title", "color"])
        return mapping
    }
    
    override init () {
    }
    
    init(title:String, color:String) {
        self.title = title
        self.color = color
    }
    
    convenience init(unboxer:Unboxer) throws {
        self.init(
            title: try unboxer.unbox(keyPath: "title"),
            color: try unboxer.unbox(keyPath: "color")
        )
    }
}
